data/src/client.rs

```diff
@@ -243,7 +334,72 @@ impl Client {
+                    if let Some(ChatHistoryRequest {
+                        subcommand,
+                        message_reference,
+                        ..
+                    }) = self.chathistory_requests.get(&target)
+                    {
+                        if Some(User::from(Nick::from("HistServ"))) == message.user() {
+                            // HistServ provides event-playback without event-playback
+                            // which would require client-side parsing to map appropriately.
+                            // Avoid that complexity by only providing that functionality
+                            // via event-playback.
+                            vec![]
+                        } else {
+                            match &message.command {
+                                Command::NICK(_) => {
+                                    let target = message::Target::Channel {
+                                        channel: target,
+                                        source: source::Source::Server(None),
+                                    };
+
+                                    vec![Event::ChatHistoryWithTarget(
+                                        message,
+                                        self.nickname().to_owned(),
+                                        target,
+                                        subcommand.clone(),
+                                        message_reference.clone(),
+                                    )]
+                                }
+                                Command::QUIT(_) => {
+                                    let target = message::Target::Channel {
+                                        channel: target,
+                                        source: source::Source::Server(Some(source::Server::new(
+                                            source::server::Kind::Quit,
+                                            message
+                                                .user()
+                                                .map(|user| Nick::from(user.nickname().as_ref())),
+                                        ))),
+                                    };
+
+                                    vec![Event::ChatHistoryWithTarget(
+                                        message,
+                                        self.nickname().to_owned(),
+                                        target,
+                                        subcommand.clone(),
+                                        message_reference.clone(),
+                                    )]
+                                }
+                                _ => vec![Event::ChatHistorySingle(
+                                    message,
+                                    self.nickname().to_owned(),
+                                    subcommand.clone(),
+                                    message_reference.clone(),
+                                )],
```

Comment:
I assume we're just forwarding all messages within here to history and not calling `handle` as we don't want to update local state based on history? That makes sense.

Reply:
Correct.  Particularly since we're requesting `draft/event-playback` when available there can be messages that would mess with local state.